backend/src/main/java/saci/domain/service/RoleService.java

```diff
@@ -28,10 +32,12 @@ public List<Role> getRoles() {
     }
 
     public Role editRole(Long roleId, Role updatedRole) {
+        String errorMessage = "Role not found";
+        log.error(errorMessage);
```

Comment:
This will log everytime, you need to put it inside the throw, when the error actually happens. Like you did with other cases.